Userland/Libraries/LibVideo/Color/TransferCharacteristics.cpp

```diff
@@ -0,0 +1,124 @@
+    case TransferCharacteristics::BT2020BitDepth12:
+        // https://en.wikipedia.org/wiki/Rec._601#Transfer_characteristics
+        // https://en.wikipedia.org/wiki/Rec._709#Transfer_characteristics
+        // https://en.wikipedia.org/wiki/Rec._2020#Transfer_characteristics
+        // These three share identical OETFs.
+        if (value < bt_601_beta * bt_601_linear_coef)
+            return value / bt_601_linear_coef;
+        return powf((value + (bt_601_alpha - 1.0f)) / bt_601_alpha, 1.0f / bt_601_gamma);
+    case TransferCharacteristics::SRGB:
+        // https://color.org/sRGB.pdf
+        if (value < srgb_inverse_linear_coef * srgb_inverse_beta)
+            return value / srgb_inverse_linear_coef;
+        return powf((value + (srgb_alpha - 1.0f)) / srgb_alpha, srgb_gamma);
+    case TransferCharacteristics::SMPTE2084: {
+        // https://en.wikipedia.org/wiki/Perceptual_quantizer
+        auto gamma_adjusted = powf(value, 1.0f / pq_m2);
+        auto numerator = max(gamma_adjusted - pq_c1, 0.0f);
+        auto denominator = pq_c2 - pq_c3 * gamma_adjusted;
+        return powf(numerator / denominator, 1.0f / pq_m1) * (pq_max_luminance / sdr_max_luminance);
+    }
+    case TransferCharacteristics::HLG:
+        // https://en.wikipedia.org/wiki/Hybrid_log-gamma
+        if (value < 0.5f)
+            return (value * value) / 3.0f;
+        return (expf((value - hlg_c) / hlg_a) + hlg_b) / 12.0f;
+    default:
+        dbgln("Unsupported transfer function {}", static_cast<u8>(transfer_function));
+        VERIFY_NOT_REACHED();
+    }
+}
+
+float TransferCharacteristicsConversion::to_non_linear_luminance(float value, TransferCharacteristics transfer_function)
+{
+    switch (transfer_function) {
+    case TransferCharacteristics::BT709:
+    case TransferCharacteristics::BT601:
+    case TransferCharacteristics::BT2020BitDepth10:
+    case TransferCharacteristics::BT2020BitDepth12:
+        // https://en.wikipedia.org/wiki/Rec._601#Transfer_characteristics
+        // https://en.wikipedia.org/wiki/Rec._709#Transfer_characteristics
+        // https://en.wikipedia.org/wiki/Rec._2020#Transfer_characteristics
+        // These three share identical OETFs.
+        if (value < bt_601_beta)
+            return bt_601_linear_coef * value;
+        return bt_601_alpha * powf(value, bt_601_gamma) - (bt_601_alpha - 1.0f);
+    case TransferCharacteristics::SRGB:
+        // https://color.org/sRGB.pdf
+        if (value < srgb_inverse_beta)
+            return value * srgb_inverse_linear_coef;
+        return srgb_alpha * powf(value, 1.0f / srgb_gamma) - (srgb_alpha - 1.0f);
+    case TransferCharacteristics::SMPTE2084: {
+        // https://en.wikipedia.org/wiki/Perceptual_quantizer
+        auto linear_value = powf(value * (sdr_max_luminance / pq_max_luminance), pq_m1);
+        auto numerator = pq_c1 + pq_c2 * linear_value;
+        auto denominator = 1 + pq_c3 * linear_value;
+        return pow(numerator / denominator, pq_m2);
```

Comment:
(See above)
Like here you missed the trailing `f` which would not be needed with the AK version

Reply:
Good point, ty